### PR TITLE
Add codeowners support for repo

### PR DESCRIPTION
**Pull request by ryanbogan**
Signed-off-by: Ryan Bogan <rbogan@amazon.com>

### Description
Add codeowners support for repo

### Issues resolved
Progress on:
[Meta issue](https://github.com/opensearch-project/project-meta/issues/31)
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by CEHENKLE — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by CEHENKLE** [comment]
@opensearch-project/common-utils I've updated your branch protection rules to point to this codeowners file.  Do you want the number of PRs set to 1 or 2?

**Comment by qreshi** [comment]
> @opensearch-project/common-utils I've updated your branch protection rules to point to this codeowners file. Do you want the number of PRs set to 1 or 2?

By number of PRs do you mean the reviewer approval required to merge? If so, I'd prefer 2 to be consistent with the other repos.

**Comment by CEHENKLE** [comment]
@qreshi Sorry, that's what I get for doing multiple things at once.  DOH!  ;) 

Yes, the question should be:  How *approvals* do you want to require -- specifically the "number of approvals and no changes requested before they can be merged."

At some point we may require consistency, but I don't think the direction is clear.  So since I'm having to go in and update the rules, if you have a preference, LMK.

**Pull request merged**
